src/ProductConstructionService/ProductConstructionService.Api/Configuration/ApiRedirection.cs

```diff
@@ -0,0 +1,191 @@
+        {
+            foreach (var (key, values) in context.Request.Headers)
+            {
+                switch (key.ToLower())
+                {
+                    // We shouldn't copy any of these request headers
+                    case "host":
+                    case "authorization":
+                    case "cookie":
+                    case "content-length":
+                    case "content-type":
+                        continue;
+                    default:
+                        try
+                        {
+                            req.Headers.Add(key, values.ToArray());
+                        }
+                        catch
+                        {
+                            // Some headers set by the client might be invalid (e.g. contain :)
+                        }
+                        break;
+                }
+            }
+
+            configureRequest(req);
+
+            HttpResponseMessage res = await client.SendAsync(req, HttpCompletionOption.ResponseHeadersRead);
+            context.Response.RegisterForDispose(res);
+
+            foreach (var (key, values) in res.Headers)
+            {
+                switch (key.ToLower())
+                {
+                    // Remove headers that the response doesn't need
+                    case "set-cookie":
+                    case "x-powered-by":
+                    case "x-aspnet-version":
+                    case "server":
+                    case "transfer-encoding":
+                    case "access-control-expose-headers":
+                    case "access-control-allow-origin":
+                        continue;
+                    default:
+                        if (!context.Response.Headers.ContainsKey(key))
+                        {
+                            context.Response.Headers.Append(key, values.ToArray());
+                        }
+
+                        break;
+                }
+            }
+
+            context.Response.StatusCode = (int)res.StatusCode;
+            if (res.Content != null)
+            {
+                foreach (var (key, values) in res.Content.Headers)
+                {
+                    if (!context.Response.Headers.ContainsKey(key))
+                    {
+                        context.Response.Headers.Append(key, values.ToArray());
+                    }
+                }
+
+                using (var data = await res.Content.ReadAsStreamAsync())
+                {
+                    await data.CopyToAsync(context.Response.Body);
+                }
+            }
+
+            return new EmptyResult();
+        }
+    }
+
+    public static async Task RedirectionHandler(HttpContext ctx, string apiRedirectionTarget)
+    {
+        var logger = ctx.RequestServices.GetRequiredService<ILogger<IApplicationBuilder>>();
+        logger.LogDebug("Preparing for redirect to '{redirectUrl}'", apiRedirectionTarget);
```

Comment:
This log doesn't look very useful in this form, maybe it'd be better if it had the source and target url?

Reply:
We see that in a previous log statement (if we enable debug logs)